micro-infra-spring-base/src/main/groovy/com/ofg/infrastructure/web/correlationid/CorrelationIdAspect.java

```diff
@@ -80,6 +90,17 @@ public Object call() throws Exception {
         });
     }
 
+    @Around("anyControllerOrRestControllerWithPublicWebAsyncTaskMethod()")
+    public Object wrapWebAsyncTaskWithCorrelationId(ProceedingJoinPoint pjp) throws Throwable {
+        final WebAsyncTask webAsyncTask = (WebAsyncTask) pjp.proceed();
+        log.debug("Wrapping webAsyncTask with correlation id [" + CorrelationIdHolder.get() + "]");
+
+        Field callableField = WebAsyncTask.class.getDeclaredField("callable");
```

Comment:
Actually, what you want to do is:

    new WebAsyncTask(
      webAsyncTask.getTimeout(),
      webAsyncTask.getAsyncTaskExecutor(),
      CorrelationIdUpdater.wrapCallableWithId(webAsyncTask.getCallable()))     
    );

However `WebAsyncTask` can also have an `executor` name (`org.springframework.web.context.request.async.WebAsyncTask#WebAsyncTask(java.lang.Long, java.lang.String, java.util.concurrent.Callable<V>)`) - so you actually have to deal with both cases. IMHO it's still better than reflection.